### PR TITLE
transform zone to zones

### PR DESCRIPTION
**Pull request by zishen**
What type of PR is this?

/kind cleanup

What this PR does / why we need it:

transform zone to zones

Which issue(s) this PR fixes:
part of https://github.com/karmada-io/karmada/issues/3952

Special notes for your reviewer:

@XiShanYongYe-Chang @whitewindmills

Does this PR introduce a user-facing change?:

none

### CODE REVIEW

**Comment by zishen** [comment]
@whitewindmills @RainbowMango 

I cleared the zones when I set them because they can't exist at the same time.

I am not sure whether this change is reasonable. Could you please confirm it according to the business?

File: pkg/apis/cluster/mutation/mutation.go [hunk 1]
```diff
@@ -18,11 +18,20 @@ const (
...
 func MutateCluster(cluster *clusterapis.Cluster) {
-	MutateClusterTaints(cluster.Spec.Taints)
+	mutateClusterTaints(cluster.Spec.Taints)
+	mutateClusterZone(cluster)
 }
 
-// MutateClusterTaints add TimeAdded field for cluster NoExecute taints only if TimeAdded not set.
-func MutateClusterTaints(taints []corev1.Taint) {
+// mutateClusterZone add zones field for cluster if Zones not set but Zone set only.
+func mutateClusterZone(cluster *clusterapis.Cluster) {
```
Comment (whitewindmills) [inline pkg/apis/cluster/mutation/mutation.go:26]: could you use a more appropriate function name? such as `setDefaultClusterZones`.

**Comment by whitewindmills** [comment]
> I am not sure whether this change is reasonable. Could you please confirm it according to the business?

I think it's okay.

File: pkg/apis/cluster/mutation/mutation.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/apis/cluster/mutation/mutation.go:26]: How about the `MigrateZoneToZones()`, no big deal since it will be removed eventually.

Comment (whitewindmills) [reply pkg/apis/cluster/mutation/mutation.go:26]: it's okay 😃

**Comment by RainbowMango** [comment]
> I cleared the zones when I set them because they can't exist at the same time.

I guess you mean `cleared the .spec.zone` during migration. 
Yes, I think we should do that.

**Review by whitewindmills — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**